src/index.ts

```diff
@@ -44,23 +46,118 @@ export default class WorkspaceClient {
         return new RemoteAPI(resources);
     }
 
-    public static getJsonRpcApi(entryPoint: string): IWorkspaceMasterApi {
-        const transport = new WebSocketClient();
-        return new WorkspaceMasterApi(transport, entryPoint);
-    }
-
     private static createAxiosInstance(config: IRestAPIConfig): AxiosInstance {
-        if (config.ssCrtPath && this.isItNode() && fs.existsSync(config.ssCrtPath)) {
-            const agent = new https.Agent({
-                ca: fs.readFileSync(config.ssCrtPath)
-            });
-            return axios.create({httpsAgent: agent});
+        if (!this.isItNode()) {
+            this.addLogInterceptorsIfEnabled(axios, config);
+            return axios;
         }
 
+        const certificateAuthority = this.getCertificateAuthority(config);
+        const proxyUrl = process.env.http_proxy;
+        if (proxyUrl && proxyUrl !== '' && config.baseUrl) {
+            const parsedBaseUrl = url.parse(config.baseUrl);
+            if (parsedBaseUrl.hostname && this.shouldProxy(parsedBaseUrl.hostname)) {
+                const axiosRequestConfig: AxiosRequestConfig | undefined = {
+                    proxy: false,
+                };
+                const parsedProxyUrl = url.parse(proxyUrl);
+                const mainProxyOptions = this.getMainProxyOptions(parsedProxyUrl);
+                const httpsProxyOptions = this.getHttpsProxyOptions(mainProxyOptions, parsedBaseUrl.hostname, certificateAuthority);
+                const httpOverHttpsAgent = tunnel.httpOverHttps({ proxy: httpsProxyOptions });
+                const httpsOverHttpsAgent = tunnel.httpsOverHttps(httpsProxyOptions);
+                const urlIsHttps = (parsedBaseUrl.protocol || 'http:').startsWith('https:');
+                const proxyIsHttps = (parsedProxyUrl.protocol || 'http:').startsWith('https:');
+                if (urlIsHttps) {
+                    if (proxyIsHttps) {
+                        axiosRequestConfig.httpsAgent = httpsOverHttpsAgent;
+                    } else {
+                        axiosRequestConfig.httpsAgent = tunnel.httpsOverHttp({
+                            proxy: mainProxyOptions,
+                            ca: certificateAuthority ? [certificateAuthority] : undefined
+                        });
+                    }
+                } else {
+                    axiosRequestConfig.httpAgent = proxyIsHttps ? httpsOverHttpsAgent : httpOverHttpsAgent;
```

Comment:
afaict here it should be:
```typescript
axiosRequestConfig.httpAgent = proxyIsHttps ? httpOverHttpsAgent : httpOverHttpAgent,
```

Of course you would have to define them just above.

TBH I'd rather define all the const just above as you did for `httpOverHttpsAgent`. To me it would make the logic much clearer and voir this type of errors

Reply:
fixed